### PR TITLE
API set up

### PR DESCRIPTION
**Pull request by darthrighteous**
### **What does this PR do?**
Have CRUD API endpoints set up.

### **Description of Tasks to be completed**
Have the following endpoints, validation logic and tests for them working:

1. `GET /v1/questions?sortBy=score` // retrieves the list of questions (optional score sort query)
2. `GET /v1/questions/:questionId` // retrieves a specific question
3. `POST /v1/questions` // posts a question
4. `POST /v1/questions/:questionId/answers` // posts an answer to a question
5. `PATCH /v1/questions/:questionId/answers/:answerId` // accepts an answer
6. `PATCH /v1/questions/:questionId` // upvotes/downvotes or edits a question's details
7. `DELETE /v1/questions/:questionId` // deletes a quesition

> App hosted on heroku at https://fast-river-42221.herokuapp.com/

### **How should this be manually tested**
- Clone the repository, cd into it, checkout feature branch
- Run the following `npm run build` `npm run serve` `npm run start`
- Use postman to test all end points using http://localhost:4001/ (e.g for endpoint 2 use http://localhost:4001/v1/questions/1)

 **some endpoints require request body input**
- endpoint 3/4 - JSON request body 

> example

```
{
	"title": "dummy title",
	"body": "dummy body",
	"date": "dummy date",
	"user": "darth"
}
```
- endpoint 6 - JSON request body  (**attribute** can be '_score_', _'title'_, _'body'_)
    - if **attribute** is _score_, then _value_ can be '_increment_' or '_decrement_' to upvote/downvote respectively.
    - if **attribute** is title or body, value should be whatever new string to replace.

> example

```
{
	"attribute": "score",
	"value": "increment"
}
```
### **What are relevant pivotal tracker stories?**
#159784977 #159784908 #159784912 #159784920 #159784926 #159784945 #159784945

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import Answer from './models/answer';
+import Question from './models/question';
+import Comment from './models/comment';
+
+const q1 = {
+  title: 'Dummy Question',
+  body: 'How do you divide a prime number by itself?',
+  date: '20/03/2003',
+  user: 'tyronne',
+};
```
Comment (Billmike) [inline utils.js]: New line after each object declaration.

**Review by Billmike — CHANGES_REQUESTED** [review]

**Comment by Billmike** [comment]
1. Fix the merge conflicts you have.
2. 87 commits?? 😱

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "presets": ["env"] }
```
[the remark this answers is not in the document]
Comment (Darthrighteous) [reply .babelrc]: done

File: .coveralls.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+repo_token: qjtcWieTaMQEznKQOpRAIkz6xSaBFeq2Y
```
[the remark this answers is not in the document]
Comment (Darthrighteous) [reply .coveralls.yml]: done

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+	"extends": "airbnb-base"
+}
```
[the remark this answers is not in the document]
Comment (Darthrighteous) [reply .eslintrc]: done

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/node_modules
```
[the remark this answers is not in the document]
Comment (Darthrighteous) [reply .gitignore]: done

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+language: node_js
+node_js: 
+  - "stable"
+cache:
+  directories:
+    - "node_modules"
+
+after_success:
+  - npm run coverage
```
[the remark this answers is not in the document]
Comment (Darthrighteous) [reply .travis.yml]: done

File: utils.js (hunk 1 above)
Comment (Darthrighteous) [reply utils.js]: done

**Comment by Darthrighteous** [comment]
@Billmike Thanks for the review. I think I've made all the corrections pointed out. 
Please let me know if I've miss anything!

**Comment by Darthrighteous** [comment]
Added heroku link

**Comment by Darthrighteous** [comment]
I have implemented es-lint rules for valid-jsdoc and max-len.

**Pull request closed without merging**